.github/workflows/dev.yaml

```diff
@@ -11,37 +8,48 @@ on:
   workflow_dispatch:
 
 jobs:
-  build:
+  build-and-test:
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-node@v4
         with:
-          node-version: 'lts/iron'
+          node-version: 'lts/jod'
       - uses: actions/cache@v4
         id: npm-cache
         with:
           path: '**/node_modules'
           key: ${{ runner.os }}-modules-${{ hashFiles('**/package-lock.json') }}
+
+      # Project  Setup
       - name: Install dependencies
         if: steps.npm-cache.outputs.cache-hit != 'true'
         run: npm ci
-      - run: npm run build --if-present
       - name: Linting files
         run: npm run lint
-      - uses: saucelabs/sauce-connect-action@v1
+
+      # Start Sauce Connect tunnel for SauceLabs
+      - name: Start Sauce Connect
+        uses: saucelabs/sauce-connect-action@v2
         with:
           username: ${{ secrets.SAUCE_USERNAME }}
           accessKey: ${{ secrets.SAUCE_ACCESS_KEY }}
-          noSSLBumpDomains: all
-          noProxyCaching: true
           tunnelIdentifier: github-action-tunnel
-      - run: npm test
+
+      # Unit Tests Step
+      - name: Run unit tests
+        run: npm run test:unit:ci:pull:request
         env:
-          SAUCE_USERNAME: ${{secrets.SAUCE_USERNAME}}
-          SAUCE_ACCESS_KEY: ${{secrets.SAUCE_ACCESS_KEY}}
-      - name: Coveralls
-        uses: coverallsapp/github-action@master
-        with:
-          github-token: ${{ secrets.GITHUB_TOKEN }}
+          SAUCE_USERNAME: ${{ secrets.SAUCE_USERNAME }}
+          SAUCE_ACCESS_KEY: ${{ secrets.SAUCE_ACCESS_KEY }}
+      ## Coveralls down.
```

Comment:
TODO: uncomment this, but coveralls was down when i was trying to verify integration tests. Don't merge with this commented.

<img width="368" height="139" alt="Screenshot 2025-10-03 at 4 53 21 PM" src="https://github.com/user-attachments/assets/50cd0eda-d007-4abe-88c5-2e96da291c75" />